libavpipe/src/avpipe_mux.c

```diff
@@ -379,20 +378,20 @@ get_next_packet(
         if (pkts[index].pts == pkt->pts)
             goto read_frame_again;
         xctx->is_pkt_valid[index] = 1;
-        if (pkt->pts >= 0) {
+        if (pkt->dts >= 0) {
             if (index == 0) {
-                if (pkt->pts > in_mux_ctx->last_video_pts)
-                    in_mux_ctx->last_video_pts = pkt->pts;
+                if (pkt->dts > in_mux_ctx->last_video_dts)
+                    in_mux_ctx->last_video_dts = pkt->dts;
                 else {
-                    pkt->pts += in_mux_ctx->last_video_pts;
```

Comment:
Looking at the original code - Maybe I'm imagining things, but shouldn't this add the duration of the packet instead of the PTS? That is, adding the `last_video_pts` looks like it would double the PTS value ...

Reply:
Ah. So I guess this causes the PTS doubling in the issue description. That has to be a bug right? It probably should be something like:
```go
pkt->pts = in_mux_ctx->last_video_pts + duration
```